crates/packaging/src/https.rs

```diff
@@ -56,6 +67,13 @@ impl<'a> PackageMetadata<'a> {
             }
         };
 
+        // Next, check if there are misleading characters in the URL
+        for misleading_character in MISLEADING_CHARACTERS_IN_URL {
+            if url.contains(misleading_character) {
+                return Err(UrlProblem::MisleadingCharacter);
```

Comment:
I'd expect this to run faster if we iterated over the url's characters, and then for each one, checked to see if it was one of the misleading ones. To do that in Rust, I think you'd want something like:

```rust
if url.chars().any(|ch| MISLEADING_CHARACTERS_IN_URL.contains(&ch)) {
    return Err(UrlProblem::MisleadingCharacter);
}
```

Reply:
Thank you for this tip, I implemented it 🙏